replication/parser.go

```diff
@@ -40,6 +40,8 @@ type BinlogParser struct {
 	useDecimal          bool
 	ignoreJSONDecodeErr bool
 	verifyChecksum      bool
+
+	rowsEventDecodeFunc func(*RowsEvent, []byte) error
```

Comment:
maybe we can set the default value to `(*RowsEvent).Decode`, so in `parseEvent` we don't need to check it many times.